core/pom.xml

```diff
@@ -88,6 +88,11 @@
       <artifactId>jaxb-api</artifactId>
       <optional>true</optional>
     </dependency>
+    <dependency>
+      <groupId>io.smallrye.config</groupId>
+      <artifactId>smallrye-config</artifactId>
+      <optional>true</optional>
```

Comment:
What is the motivation of marking this dependency as optional here? The `AbstractConfig` class, requiring this dependency, looks like an important part of the hono-core module.

Reply:
Not necessarily. If you use Spring Boot based configuration, then you so not need it. Once we have removed all Spring Boot based code/components, we can/should make it a required dependency ...

Reply:
But as I understand it, the `hono-service-base-spring` module would still need this dependency because the `AbstractConfig` class is used there (e.g. using the `ServerConfig` sub class of `AbstractConfig` being used in the `hono-service-base-spring` `AbstractAdapterConfig` class).

Reply:
As far as I can tell, there is no *AbstractAdapterConfig* class in the *hono-service-base-spring* module.

Reply:
Sorry, I meant `AbstractAdapterConfig` in `hono-adapter-base-spring`. 
The point is that I think in practise, the spring modules still need the `smallrye-config` dependency via usage of the `AbstractConfig` class.

Reply:
Why do you think so? FMPOV no classes from smallrye-config will be needed/used during runtime in a Spring Boot based component, or am I mistaken?

Reply:
As I see it, during runtime the `AbstractConfig` class needs to be loaded by the Spring Boot based components. And this means the `smallrye-config` classes will be loaded as well (via the `GenericOptions` class used in the corresponding `AbstractConfig` constructor).
You could for example delete the `AbstractConfig` class and see that there is a compile error in `hono-service-base` `AbstractServiceBase` and in `hono-adapter-base-spring` `AbstractAdapterConfig`, for example.

Reply:
It is true that the AbstractConfig class is being loaded but that doesn't mean that all classes that have been required to compile it are being loaded as well. If the constructor accepting the GenericOptions argument is never invoked, then the GenericOptions class will not be loaded. As the Spring Boot based components to not bind any properties to the *Options classes, these constructors are never invoked. This is also consistent with the fact that the integration tests using the Spring Boot based components succeed but none of these components' Docker containers contain the *smallrye-config* artifact ...